component/configsource.go

```diff
@@ -0,0 +1,37 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package component // import "go.opentelemetry.io/collector/component"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/configmapprovider"
+)
+
+// ConfigSource is a component that provides a configuration map or value.
+type ConfigSource interface {
+	configmapprovider.Provider
```

Comment:
Why do you need a ConfigSource and a Provider if they are the same? This is what I was suggesting https://github.com/open-telemetry/opentelemetry-collector/issues/3924. I think we should have only one concept.